service_discovery/svc_utils/services.go

```diff
@@ -87,38 +87,46 @@ func IsObjectInClustersetFilter(cname, ns, obj string) bool {
 	return false
 }
 
+func IsSvcPortInClustersetFilter(cname, ns, obj string, port int32) bool {
+	nsSvcFilter, present := clustersetServiceFilter[cname]
```

Comment:
We may need a lock here if clustersetServiceFilter gets accessed by multiple object update.